btc_light_client.go

```diff
@@ -46,25 +47,51 @@ func (lc *BTCLightClient) FindPreviousCheckpoint() (blockchain.HeaderCtx, error)
 	return nil, nil
 }
 
-func (lc *BTCLightClient) AddHeader(height int64, header wire.BlockHeader) error {
-	return lc.btcStore.AddHeader(height, header)
+func (lc *BTCLightClient) SetHeader(height int64, header wire.BlockHeader) error {
+	return lc.btcStore.SetHeader(height, header)
 }
 
 // We assume we always insert valid header. Acctually, Cosmos can revert a state
 // when module return error so this assumtion is reasonable
 func (lc *BTCLightClient) InsertHeaders(headers []wire.BlockHeader) error {
 	latestHeight := lc.btcStore.LatestHeight()
+
+	return lc.insertHeadersWithPosition(uint64(latestHeight), headers)
+}
+
+func (lc *BTCLightClient) insertHeadersWithPosition(height uint64, headers []wire.BlockHeader) error {
+	insertHeight := height + 1
 	for _, header := range headers {
 		if err := lc.CheckHeader(header); err != nil {
 			return err
 		}
 
-		latestHeight = latestHeight + 1
-		lc.AddHeader(latestHeight, header)
+		lc.SetHeader(int64(insertHeight), header)
+		insertHeight = insertHeight + 1
 	}
 	return nil
 }
 
+func (lc *BTCLightClient) HandleFork(headers []wire.BlockHeader) error {
```

Comment:
this is how it should work:
1. Firstly we check the total difficulty of the fork that is already recorded
2. then we check if the new for total difficulty is bigger
3. if yes, then we update to the new fork

Reply:
Yes, I do the same thing in this PR. A total difficulty in our context is `TotalWork` computed from sum of header.Bits in this fork.